client/asset/eth/multirpc.go

```diff
@@ -569,12 +569,82 @@ func (m *multiRPCClient) voidUnusedNonce() {
 	m.lastNonce.stamp = time.Time{}
 }
 
-func (m *multiRPCClient) reconfigure(ctx context.Context, settings map[string]string) error {
+// createAndCheckProviders creates and connects to providers. It checks that
+// unknown providers have a sufficient api to trade and saves good providers to
+// file. One bad provider or connect problem will cause this to error.
+func createAndCheckProviders(walletDir string, endpoints []string, net dex.Network,
+	log dex.Logger, ctx context.Context) error {
+	var localCP map[string]bool
+	path := filepath.Join(walletDir, "compliant-providers.json")
+	b, err := os.ReadFile(path)
+	if err == nil {
+		if err := json.Unmarshal(b, &localCP); err != nil {
+			log.Warnf("Couldn't parse compliant providers file: %v", err)
+		}
+	}
+	if localCP == nil {
+		localCP = make(map[string]bool)
+	}
+
+	var writeLocalCP bool
+
+	var unknownEndpoints []string
+	for _, p := range endpoints {
+		d := domain(p)
+		if localCP[d] {
+			continue
+		}
+		writeLocalCP = true
+		localCP[d] = true
+		if _, known := compliantProviders[d]; known {
+			continue
+		}
+		unknownEndpoints = append(unknownEndpoints, p)
+	}
+
+	providers := []*provider{}
+	if len(unknownEndpoints) > 0 {
+		ctx, cancel := context.WithTimeout(ctx, time.Second*10)
```

Comment:
I wonder if 10 seconds is long enough if there are multiple providers to check.

Reply:
Multiply timeout by number of providers?

Reply:
There is already a per-connection default timeout in `connectProviders`. I think what needs to happen is to change the call signature of `rpcTest.f` to `func(context.Context, *provider) error`, and create a new timed context for each request in `checkProvidersCompliance`.

Reply:
Changed the `rpcTest.f` function signature as suggested.

Reply:
Forgot to remove this ctx it seems.